pkg/apis/configuration/validation/validation_test.go

```diff
@@ -1620,3 +1652,210 @@ func TestValidateTimeFails(t *testing.T) {
 		t.Errorf("validateTime returned no errors for invalid input %v", time)
 	}
 }
+
+func TestValidateUpstreamHealthCheck(t *testing.T) {
+	hc := &v1alpha1.HealthCheck{
+		Enable:   true,
+		Path:     "/healthz",
+		Interval: "4s",
+		Jitter:   "2s",
+		Fails:    3,
+		Passes:   2,
+		Port:     8080,
+		TLS: &v1alpha1.UpstreamTLS{
+			Enable: true,
+		},
+		ConnectTimeout: "1s",
+		ReadTimeout:    "1s",
+		SendTimeout:    "1s",
+		Headers: []v1alpha1.Header{
+			{
+				Name:  "Host",
+				Value: "my.service",
+			},
+		},
+		StatusMatch: "! 500",
+	}
+
+	allErrs := validateUpstreamHealthCheck(hc, field.NewPath("healthCheck"))
+
+	if len(allErrs) != 0 {
+		t.Errorf("validateUpstreamHealthCheck returned errors for valid input %v", hc)
```

Comment:
small comment about consistency for all added test functions. In existing test functions in error messages we use `function()` - i.e. with the name of the function and parentheses.

Reply:
They are mixed actually, there's no consistency in the whole file